client-app/ui-kit/components/atoms/checkbox/vc-checkbox.vue

```diff
@@ -22,59 +22,53 @@
   </label>
 </template>
 
-<script setup lang="ts">
+<script setup lang="ts" generic="V, T extends boolean | V[]">
+import { isBoolean } from "lodash";
 import { computed } from "vue";
 
-const emit = defineEmits<{
-  (event: "update:modelValue", value: boolean | any[]): void;
-  (event: "change", value: boolean | any[]): void;
-}>();
-
-const props = defineProps({
-  disabled: Boolean,
+export interface IEmits<T> {
+  (event: "update:modelValue", value: T): void;
+  (event: "change", value: T): void;
+}
 
-  modelValue: {
-    type: [Boolean, Array],
-    default: () => [],
-  },
+export interface IProps<V, T extends boolean | V[]> {
+  modelValue?: T;
+  value?: T extends boolean ? never : V;
+  disabled?: boolean;
+  name?: string;
+}
 
-  name: {
-    type: String,
-    default: null,
-  },
+const emit = defineEmits<IEmits<T>>();
 
-  // only for array of model value
-  value: {
-    type: [Boolean, String, Number, Object],
-    default: undefined,
-  },
-});
+const props = defineProps<IProps<V, T>>();
 
 const checked = computed<boolean>(() =>
-  typeof props.modelValue === "boolean" ? props.modelValue : props.modelValue.includes(props.value),
+  isBoolean(props.modelValue) ? props.modelValue : props.modelValue?.includes(props.value!) ?? false,
 );
 
 function change() {
   if (props.disabled) {
     return;
   }
 
-  if (typeof props.modelValue === "boolean") {
-    const newValue = !props.modelValue;
-    emit("update:modelValue", newValue);
-    emit("change", newValue);
+  let newValue: T;
+
+  if (isBoolean(props.modelValue)) {
+    newValue = !props.modelValue as T;
```

Comment:
else if (Array.isArray(props.modelValue))

Reply:
Generic are guarantee it can be only boolean or array, so reduntant

Reply:
Reverted